src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        currentState = { ...currentState, ...action.extraData };
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (key in currentState) {
+            const { [key]: _, ...rest } = currentState;
```

Comment:
The variable name '_' is not descriptive. When destructuring for the purpose of omitting properties, it's better to use a more descriptive name, like 'unusedVariable', to improve code readability.